scripts/library.sh

```diff
@@ -356,16 +358,30 @@ function lint_markdown() {
   run_lint_tool mdl "linting markdown files" "-r ~MD013" $@
 }
 
-# Return 0 if the given parameter is an integer, otherwise 1
-# Parameters: $1 - an integer
+# Return whether the given parameter is an integer.
+# Parameters: $1 - integer to check
 function is_int() {
   [[ -n $1 && $1 =~ ^[0-9]+$ ]]
 }
 
-# Return 0 if the given parameter is the knative release/nightly gcr, 1
-# otherwise
-# Parameters: $1 - gcr name, e.g. gcr.io/knative-nightly
+# Return whether the given parameter is the knative release/nightly GCF.
+# Parameters: $1 - full GCR name, e.g. gcr.io/knative-foo-bar
 function is_protected_gcr() {
   [[ -n $1 && "$1" =~ "^gcr.io/knative-(releases|nightly)/?$" ]]
 }
 
+# Returns the canonical path of a filesystem object.
+# Parameters: $1 - path to return in canonical form
+#             $2 - base dir for relative links; optional, defaults to current
+function get_canonical_path() {
+  # We don't use readlink because it's not available on every platform.
+  local path=$1
+  local pwd=${2:-.}
+  [[ ${path} == /* ]] || path="${pwd}/${path}"
```

Comment:
would this left statement ever be false?

Reply:
It's false for all relative paths.

Reply:
oh right, misread it, thought it was `.*` initially